dcos/recordio.py

```diff
@@ -0,0 +1,164 @@
+
+5\n
+hello
+6\n
+world!
+
+Note that this currently only supports record lengths
+encoded as base 10 integer values with newlines as a
+delimiter. This is to provide better language portability
+portability: parsing a base 10 integer is simple. Most
+other "Record-IO" implementations use a fixed-size header
+of 4 bytes to directly encode an unsigned 32 bit length.
+"""
+
+from dcos.errors import DCOSException
+
+
+class Encoder(object):
+    """Encode an arbitray message type into a 'RecordIO' message.
+
+       This class encapsulates the process of encoding an
+       arbitrary string of bytes into a 'RecordIO' message. Its
+       constructor takes a serialization function of the form
+       'serialize(message)'. This serialization function is
+       responsible for knowing how to take whatever message type
+       is passed to 'encode()' and serializing it to a string.
+
+       Once 'encode(message)' is called, it will use the
+       serialization function to convert 'message' into a string
+       of bytes, wrap that string fo bytes in a 'RecordIO' frame,
+       and return it.
+    """
+
+    def __init__(self, serialize):
+        """Initializes a 'RecordIO' encoder'.
+
+        ":param serialize: a function to serialize any message
+                           passed to 'encode()'
+        type serialize: function
+        """
+
+        self.serialize = serialize
+
+    def encode(self, message):
+        """Encode a message into 'RecordIO' format.
+
+        ":param message: a message to serialize and then wrap in
+                       a 'RecordIO' frame.
+        type message: object
+        returns: a serialzied message wrapped in a 'RecordIO' frame.
+        rtype: str
+        """
+
+        s = self.serialize(message)
+        return str(len(s)) + "\n" + s
+
+
+class Decoder(object):
+    """Decode a 'RecordIO' message back to an arbitrary message type.
+
+       This class encapsulates the process of decoding a message
+       previously encoded with 'RecordIO' back to an arbitrary
+       message type. Its
+       constructor takes a deserialization function of the form
+       'deserialize(bytes)'. This deserialization function is
+       responsible for knowing how to take a fully constructed
+       'RecordIO' message containing a string of bytes and
+       deserialize it back into the original message type.
+
+       The 'decode(data)' message takes a string of bytes as
+       input and buffers them across subsequent calls to
+       construct full 'RecordIO' messages that are then decoded
+       and returned in an array.
+    """
+    HEADER = 0
+    RECORD = 1
+    FAILED = 2
+
+    def __init__(self, deserialize):
+        """Initializes a 'RecordIO' decoder'.
+
+        ":param deserialize: a function to deserialize 'RecordIO'
+                             messages built up by subsequent
+                             calls to 'decode(data)'
+        type deserialize: function
+        """
```

Comment:
Param documentation should in the doc string for the object: https://github.com/dcos/dcos-cli/blob/master/dcos/marathon.py#L63-L80. Don't ask me why. That is what the doc generating tools expect.

Also, there is an extra `"` and a missing `:`.